shell/imports/sandstorm-identicons/identicon.ts

```diff
@@ -54,14 +59,14 @@ class Identicon {
             ${rects.join("\n")}</g> </svg>`;
   }
 
-  rectangle(x, y, w, h, rectangles) {
+  rectangle(x: number, y: number, w: number, h: number, rectangles: string[]) {
     rectangles.push(`<rect x="${x}" y="${y}" width="${w}" height="${h}"/>`);
   }
 
   // adapted from: https://gist.github.com/aemkei/1325937
-  hsl2rgb(h, s, b) {
+  hsl2rgb(h: number, s: number, b: number) {
     h *= 6;
-    s = [
+    const sat = [
       b += s *= b < .5 ? b : 1 - b,
```

Comment:
I can't help but find these multiple operator-assignments on a single line terrifying

(no change needed, but yikes, would be nice if this wasn't quite so opaque)

Reply:
Yeah, I have no idea what's going on here :/